docroot/modules/humsci/hs_capx/src/Form/CapxImporterForm.php

```diff
@@ -186,11 +203,46 @@ public function save(array $form, FormStateInterface $form_state) {
         break;
 
       default:
+        $this->invalidateMigrationHashes();
         drupal_set_message($this->t('Saved the %label Capx importer.', [
           '%label' => $importer->label(),
         ]));
     }
     $form_state->setRedirectUrl($importer->toUrl('collection'));
+
+    Cache::invalidateTags(['migration_plugins', 'hs_capx_config']);
+  }
+
+  /**
+   * Invalidate migration mapper hashes to be updated on next import.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function invalidateMigrationHashes() {
+    // If the importer hasn't been executed, the table will not be created.
+    if (!$this->database->schema()->tableExists('migrate_map_hs_capx')) {
+      return;
+    }
+
+    $entity_query = $this->entityTypeManager->getStorage('node')
+      ->getQuery('OR');
+
+    // Find all node ids that art tagged with the fields. This allows us to only
+    // invalidate the hashes that are applicable.
+    foreach ($this->originalEntity->getFieldTags() as $field_name => $term_ids) {
+      foreach ($term_ids as $term_id) {
+        $entity_query->condition($field_name, $term_id);
```

Comment:
🤔 Is there any limit to the number of conditions mysql can have?

Reply:
https://stackoverflow.com/questions/4275640/mysql-in-condition-limit

doesn't appear so. in 99% of sites this will probably be < 5 terms

Reply:
Thanks, I was just looking for that documentation.